appshell/appshell_extensions_mac.mm

```diff
@@ -1175,6 +1175,152 @@ int32 SetMenuTitle(CefRefPtr<CefBrowser> browser, ExtensionString command, Exten
     return NO_ERROR;
 }
 
+OSStatus _RunToolWithAdminPrivileges( AuthorizationRef authorizationRef, const std::string& tool, const std::vector<std::string> &args)
+{
+    OSStatus status = 0;
+    if(!authorizationRef || tool.length() ==  0 || args.size() == 0)
+        return errAuthorizationInvalidSet;
+    
+    FILE *pipe = NULL;
+
+    // convert to std::vector<char *> array.
+    std::vector<char*> argv(1 + args.size(), NULL);
+
+    for (size_t i = 0; i < args.size(); ++i)
+        argv[i] = const_cast<char*>(args[i].c_str());
+    
+    // This is a deprecated API. Apple recommends a dedicated helper to
+    // fix this issue. This ideally should be replaced by SMBless API.
+    status = AuthorizationExecuteWithPrivileges(authorizationRef, tool.c_str(),
```

Comment:
This is just a stop gap solution which some other apps in the market already using. But there is no guarantee that this would ever be supported. I just upgraded my OS to 10.10.3 and thankfully it works fine there.

This is going to be a short lived one. Migration to SMBless involves much more work like creating helper tools and managing launch deamons and launchd plist. On top of this how is Brackets effected because of this change. We should take this up in the next immediate cycles.